src/network/networkd-link.c

```diff
@@ -2378,9 +2378,26 @@ static int link_set_ipv6_mtu(Link *link) {
         if (link->network->ipv6_mtu == 0)
                 return 0;
 
+        /* IPv6 protocol requires a minimum MTU of IPV6_MTU_MIN(1280) bytes
+         * on the interface. Bump up MTU bytes to IPV6_MTU_MIN. */
+        if (link->network->ipv6_mtu < IPV6_MIN_MTU) {
+
+                log_link_warning(link, "Bumping IPv6 MTU to " STRINGIFY(IPV6_MIN_MTU) ", as "
+                                 "IPv6 requires a minimum MTU of " STRINGIFY(IPV6_MIN_MTU) " bytes: %m");
+
+                link->network->ipv6_mtu = IPV6_MIN_MTU;
+        }
+
         r = sysctl_write_ip_property_uint32(AF_INET6, link->ifname, "mtu", link->network->ipv6_mtu);
-        if (r < 0)
-                log_link_warning_errno(link, r, "Cannot set IPv6 MTU for interface: %m");
+        if (r < 0) {
+                if (link->mtu < link->network->ipv6_mtu)
+                        log_link_error(link, "Cannot set IPv6 MTU %"PRIu32" higher than device MTU %"PRIu32,
+                                       link->network->ipv6_mtu, link->mtu);
```

Comment:
Please don't use "error" if the error is ignored and not propagated. `log_link_warning` is appropriate.
```C
log_link_warning(link, "Cannot set IPv6 MTU %"PRIu32" higher than device MTU %"PRIu32", ignoring.",
                            ...);
```

Reply:
ack, will update that.